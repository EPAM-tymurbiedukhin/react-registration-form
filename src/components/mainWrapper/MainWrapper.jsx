import React from 'react';

import { mainWrapper } from './mainWrapper.module.scss';

export default ({ children }) => {
  return <main className={mainWrapper}>{children}</main>;
};
