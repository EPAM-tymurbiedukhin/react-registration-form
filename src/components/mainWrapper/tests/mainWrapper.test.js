import React from 'react';
import MainWrapper from '../MainWrapper';

import { mainWrapper } from './mainWrapper.module.scss';

const StubComponent = () => <h1>Stub</h1>;

describe('<MainWrapper />', () => {
  const component = shallow(<MainWrapper children={<StubComponent />} />);

  it('should render correctly', () => {
    expect(component.html()).toMatchSnapshot();
  });

  it('should has correct className', () => {
    expect(component.hasClass(mainWrapper)).toBeTruthy();
  });

  it('should render children', () => {
    expect(component.find(StubComponent).length).toBe(1);
  });
});
