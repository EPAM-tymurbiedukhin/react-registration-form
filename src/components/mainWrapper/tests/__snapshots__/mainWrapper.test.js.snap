// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainWrapper /> should render correctly 1`] = `"<main class=\\"mainWrapper\\"><h1>Stub</h1></main>"`;
